include/swift/AST/DiagnosticsSema.def

```diff
@@ -5646,7 +5646,7 @@ NOTE(shared_state_make_immutable,none,
      "convert %0 to a 'let' constant to make 'Sendable' shared state immutable",
      (const ValueDecl *))
 NOTE(shared_state_main_actor_node,none,
-     "annotate %0 with '@MainActor' if property should only be accessed from the main actor", (const ValueDecl *))
+     "add '@MainActor' annotation to make property %0 part of global actor 'MainActor'", (const ValueDecl *))
```

Comment:
I gathered from the GitHub issue that there already is a note that says almost this exact text. It's in this same DiagnosticsSema.def file:

```
NOTE(add_globalactor_to_function,none,
     "add '@%0' to make %kind1 part of global actor %2",
     (StringRef, const ValueDecl *, Type))
```

This note already takes arguments, so we can pass in `MainActor` as both the string and the type. I think we can just remove `shared_state_main_actor_node`, and use `add_globalactor_to_function`.

Reply:
Hi, @hborla. It does make sense for me, however I do not understand how to get MainActorType to pass it as last parameter in add_globalactor_to_function. I tried getASTContext().getMainActorType() to variables in function, but it always retrun null for me. Can you give a suggestion please?

Reply:
@hborla  I tried getASTContext().getMainActorType() with different VarDecl variables and it always return null, but for functions it doesn't return null. Feels weird.
Also tried that approaches:
1) Isolation.getGlobalActor -> But the variant doesn't have any actor so I get assert
2) SwiftLang.getMainActor, but it is located in clang and I think that it shouldn't be used in AST module.

Reply:
Hmm, could you show me what you tried? This works for me locally in `ActorIsolationRequest::evaluate` at the point where it was previously diagnosing `diag::shared_state_main_actor_node` on line 5459:

```swift
          auto mainActor = ctx.getMainActorType();
          if (!mainActor)
            return isolation;

          diagVar->diagnose(diag::add_globalactor_to_function,
              mainActor->getWithoutParens().getString(), diagVar, mainActor)
              .fixItInsert(diagVar->getAttributeInsertionLoc(false),
                           diag::insert_globalactor_attr, mainActor);
```

I also needed to add a capture of `&ctx` to the lambda capture list.

Reply:
This code perfectly work. You can't imagine how many times I tried it on my local machine, but somewhy when I launch it through Xcode it's always fails(even on my second laptop). But on third laptop it works, kind of magic. Now I tried to launch tests and it works fine on all the machines
> ```swift
>           auto mainActor = ctx.getMainActorType();
>           if (!mainActor)
>             return isolation;
> 
>           diagVar->diagnose(diag::add_globalactor_to_function,
>               mainActor->getWithoutParens().getString(), diagVar, mainActor)
>               .fixItInsert(diagVar->getAttributeInsertionLoc(false),
>                            diag::insert_globalactor_attr, mainActor);
> ```

Thank you for your help @hborla